java-frontend/src/main/java/org/sonar/java/bytecode/cfg/BytecodeCFGBuilder.java

```diff
@@ -0,0 +1,234 @@
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.bytecode.cfg;
+
+import com.google.common.base.Throwables;
+import com.google.common.io.ByteStreams;
+import com.google.common.io.Closeables;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.Handle;
+import org.objectweb.asm.Label;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.util.Printer;
+import org.sonar.java.bytecode.loader.SquidClassLoader;
+import org.sonar.java.resolve.Convert;
+import org.sonar.java.resolve.Java9Support;
+import org.sonar.java.resolve.JavaSymbol;
+import org.sonar.plugins.java.api.semantic.Symbol;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class BytecodeCFGBuilder {
+
+  private BytecodeCFGBuilder() {
+  }
+
+  public static CFGBytecode buildCFG(Symbol.MethodSymbol methodSymbol, SquidClassLoader classLoader) {
+    InputStream is = classLoader.getResourceAsStream(Convert.bytecodeName(((JavaSymbol.TypeJavaSymbol) methodSymbol.owner()).getFullyQualifiedName()) + ".class");
+    try {
+      byte[] bytes = ByteStreams.toByteArray(is);
+      // to read bytecode with ASM not supporting Java 9, we will set major version to Java 8
+      if (Java9Support.isJava9Class(bytes)) {
+        Java9Support.setJava8MajorVersion(bytes);
+      }
+      ClassReader cr = new ClassReader(bytes);
+      BytecodeCFGMethodVisitor methodVisitor = new BytecodeCFGMethodVisitor(methodSymbol);
+      cr.accept(new ClassVisitor(Opcodes.ASM5) {
+        @Override
+        public MethodVisitor visitMethod(int access, String name, String desc, String signature, String[] exceptions) {
+          // TODO : name matching is not sufficient in case of overloading.
+          if (name.equals(methodSymbol.name())) {
+            return methodVisitor;
+          }
+          return null;
+        }
+      }, ClassReader.SKIP_DEBUG | ClassReader.SKIP_FRAMES);
+      return methodVisitor.cfg;
+    } catch (IOException e) {
+      throw Throwables.propagate(e);
+    } finally {
+      Closeables.closeQuietly(is);
+    }
+  }
+
+  static class CFGBytecode {
```

Comment:
`BytecodeCFG` sounds bit more natural to me, also more consisted with `BytecodeCFGBuilder`